File: src/VideoPlayer/TimeFormatter.cpp
#include "TimeFormatter.h"

#include <iomanip>
#include <sstream>

namespace videoplayer {

std::wstring FormatTime(const std::int64_t milliseconds) {
    const std::int64_t totalSeconds = milliseconds > 0 ? milliseconds / 1000 : 0;
    const std::int64_t hours = totalSeconds / 3600;
    const std::int64_t minutes = (totalSeconds / 60) % 60;
    const std::int64_t seconds = totalSeconds % 60;

    std::wostringstream text;
    text << std::setfill(L'0');
    if (hours > 0) {
        text << hours << L':' << std::setw(2) << minutes << L':' << std::setw(2) << seconds;
    } else {
        text << std::setw(2) << minutes << L':' << std::setw(2) << seconds;
    }
    return text.str();
}

}  // namespace videoplayer

File: src/VideoPlayer/Helpers/TimeFormatter.cs
using System.Globalization;

namespace VideoPlayer.Helpers;

public static class TimeFormatter
{
    public static string Format(long milliseconds)
    {
        var totalSeconds = Math.Max(0, milliseconds / 1_000);
        var hours = totalSeconds / 3_600;
        var minutes = totalSeconds % 3_600 / 60;
        var seconds = totalSeconds % 60;

        return hours > 0
            ? string.Create(
                CultureInfo.InvariantCulture,
                $"{hours}:{minutes:00}:{seconds:00}")
            : string.Create(
                CultureInfo.InvariantCulture,
                $"{minutes:00}:{seconds:00}");
    }
}
